assets/app/scripts/services/builds.js

```diff
@@ -34,13 +34,13 @@ angular.module("openshiftConsole")
       );
     };
 
-    BuildsService.prototype.cancelBuild = function(build, buildConfigName, $scope) {
+    BuildsService.prototype.cancelBuild = function(build, buildConfigName, context, $scope) {
       var canceledBuild = angular.copy(build);
       canceledBuild.status.cancelled = true;
-      DataService.update("builds", canceledBuild.metadata.name, canceledBuild, $scope).then(
+      DataService.update("builds", canceledBuild.metadata.name, canceledBuild, context).then(
         function() {
           $scope.alerts = $scope.alerts || {};
-          $scope.alerts["cancel"] = 
+          $scope.alerts["cancel"] =
```

Comment:
While we're changing all the things... If we switched these methods to return promises, we could avoid passing `$scope` to them. The caller updates the UI. Doesn't seem like the service should be doing that.